components/mqtt/publish.go

```diff
@@ -43,30 +43,51 @@ func (s *Server) PublishOnTopic(topic string, payload interface{}) {
 	s.sendMessageOnTopic(topic, jsonPayload)
 }
 
-/*
-func (s *Server) PublishMilestoneOnTopic(topic string, ms *nodebridge.Milestone) {
-	if ms == nil || ms.Milestone == nil {
+func (s *Server) PublishRawCommitmentOnTopic(topic string, commitment *iotago.Commitment) {
+	apiForVersion, err := s.NodeBridge.APIProvider().APIForVersion(commitment.Version)
+	if err != nil {
+		return
+	}
+
+	rawCommitment, err := apiForVersion.Encode(commitment)
+	if err != nil {
 		return
 	}
 
-	s.PublishOnTopicIfSubscribed(topic, &milestoneInfoPayload{
-		Index:       ms.Milestone.Index,
-		Time:        ms.Milestone.Timestamp,
-		MilestoneID: ms.MilestoneID.ToHex(),
+	s.PublishRawOnTopicIfSubscribed(topic, rawCommitment)
+}
+
+func (s *Server) PublishCommitmentInfoOnTopic(topic string, id iotago.CommitmentID) {
+	s.PublishOnTopicIfSubscribed(topic, &commitmentInfoPayload{
+		CommitmentID:    id.ToHex(),
+		CommitmentIndex: uint64(id.Index()),
 	})
 }
-*/
 
 func (s *Server) PublishBlock(blk *inx.RawBlock) {
+	version, _, err := iotago.VersionFromBytes(blk.GetData())
+	if err != nil {
+		return
+	}
 
-	block, err := blk.UnwrapBlock(serializer.DeSeriModeNoValidation, nil)
+	apiForVersion, err := s.NodeBridge.APIProvider().APIForVersion(version)
+	if err != nil {
+		return
+	}
+
+	block, err := blk.UnwrapBlock(apiForVersion)
 	if err != nil {
 		return
 	}
 
 	s.PublishRawOnTopicIfSubscribed(topicBlocks, blk.GetData())
 
-	switch payload := block.Payload.(type) {
+	basicBlk, isBasicBlk := block.Block.(*iotago.BasicBlock)
```

Comment:
I still think it will be valuable to have a topic for validator blocks, but maybe we can add that in the future?

Reply:
You're right, I always forgot about validator blocks. :stuck_out_tongue: 
Let's add that in the future.